__tests__/test-db.test.js

```diff
@@ -149,7 +144,43 @@ describe("GET /api/articles/article_id/comments", () => {
       .get("/api/articles/999/comments")
       .expect(404)
       .then((response) => {
-        expect(response.body.msg).toBe("Comments not found");
+        expect(response.body.msg).toBe("Article not found");
+      });
+  });
+  test("should return the article with code 200 if the article exists but there is no comments present", () => {
+    return request(app)
+      .get("/api/articles/4/comments")
+      .expect(200)
+      .then((result) => {
+        const body = result.body.article;
+        expect(body.length).toBe(0);
+      });
+  });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("should retrieve a post request and insert it into the comments.", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({ author: "icellusedkars", body: "This is a new comment" })
+      .expect(201);
+  });
+  test("If passed an empty field for any key, return appropriate error", () => {
+    return request(app)
+      .post("/api/articles/:article_id/comments")
+      .send({ author: "icellusedkars"})
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Invalid data input");
+      });
+  });
+  test("If passed a user that doest exist on the user database, return appropriate error", () => {
+    return request(app)
+      .post("/api/articles/:article_id/comments")
+      .send({ author: "icellars", body: "This is a new comment"})
+      .expect(401)
+      .then((response) => {
+        expect(response.body.msg).toBe("Invalid user");
```

Comment:
I see what you have done here but a 404 error would be ok here. This is because the user is "not found". Additionally PSQL will throw an error if the user doesn't exist in the DB so you could use that error that will be thrown to return the correct response - let PSQL do the work of "validating" the user/author for you.